.github/workflows/build-docker.yml

```diff
@@ -20,13 +20,13 @@ jobs:
         with:
           registry: ghcr.io
           username: ${{ github.actor }}
-          password: ${{ secrets.GITHUB_TOKEN }}
+          password: ${{ secrets.GH_PAT }}
```

Comment:
Why are you using GH_PAT here? This assumes that the user has a PAT setup?
What's wrong with using GITHUB_TOKEN?

Reply:
Hmm it was complaining before that I should not use GITHUB_TOKEN 😢 wait let me try to modify and see but last time it didn't want me to use the word GITHUB_TOKEN for some reason. So I just used GH_PAT. I'll rename it wait.

Reply:
Yeah, github complains:

![image](https://github.com/KULeuven-MICAS/snitch_cluster/assets/26665295/8a5a4463-c766-41a5-a495-ec7a99e3134e)

Reply:
@JosseVanDelm seems to build OKAY 😄  cause it logged in normally. Changed it to `GH_TOKEN` instead!

Reply:
That is so weird it totally works here:
https://github.com/KULeuven-MICAS/snax-mlir/blob/main/.github/workflows/build-docker.yml

Anyways if it works it works I guess